x/superfluid/keeper/migrate.go

```diff
@@ -329,37 +287,30 @@ func (k Keeper) validateSharesToMigrateUnlockAndExitBalancerPool(ctx sdk.Context
 
 	// Otherwise, we must ensure that the shares to migrate is less than or equal to the shares in the lock.
 	if sharesToMigrate.Amount.GT(gammSharesInLock.Amount) {
-		return sdk.Coins{}, lockuptypes.PeriodLock{}, types.MigrateMoreSharesThanLockHasError{SharesToMigrate: sharesToMigrate.Amount.String(), SharesInLock: gammSharesInLock.Amount.String()}
+		return sdk.Coins{}, types.MigrateMoreSharesThanLockHasError{SharesToMigrate: sharesToMigrate.Amount.String(), SharesInLock: gammSharesInLock.Amount.String()}
 	}
 
-	// Determine if there will be any remaining gamm shares after migration.
-	remainingGammShares := gammSharesInLock.Sub(sharesToMigrate)
-
 	// Finish unlocking directly for locked or unlocking locks
-	// This also breaks and deletes associated synthetic locks.
-	err = k.lk.ForceUnlock(ctx, *lock)
-	if err != nil {
-		return sdk.Coins{}, lockuptypes.PeriodLock{}, err
+	if sharesToMigrate.Equal(gammSharesInLock) {
+		// If migrating the entire lock:
+
+		// This breaks and deletes associated synthetic locks.
+		err = k.lk.ForceUnlock(ctx, *lock)
+		if err != nil {
+			return sdk.Coins{}, err
+		}
 	}
```

Comment:
checking: has the direct unit test been updated to account for this being conditional?

Reply:
This question led to uncovering a bug! We actually should have been doing a partialForceUnlock if the above branch is not hit. I also expanded the tests here to ensure this will get found in the future https://github.com/osmosis-labs/osmosis/pull/5162/commits/c7fd2bb93602bb9233c07a71123fd9b5031f473e